delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/serialization/EventSerializer.scala

```diff
@@ -0,0 +1,112 @@
+import ch.epfl.bluebrain.nexus.delta.sdk.model.organizations.OrganizationEvent
+import ch.epfl.bluebrain.nexus.delta.sdk.model.permissions.{Permission, PermissionsEvent}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.{PrefixIRI, ProjectEvent}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.realms.RealmEvent
+import ch.epfl.bluebrain.nexus.delta.serialization.EventSerializer._
+import io.circe.generic.extras.Configuration
+import io.circe.generic.extras.semiauto.deriveConfiguredCodec
+import io.circe.parser._
+import io.circe.syntax._
+import io.circe.{Codec, Decoder, Encoder}
+
+import scala.annotation.nowarn
+import scala.util.Try
+
+/**
+  * A json serializer for delta [[Event]] types.
+  */
+class EventSerializer extends SerializerWithStringManifest {
+
+  override def identifier: Int = 453223
+
+  override def manifest(o: AnyRef): String = o match {
+    case _: PermissionsEvent  => permissionsEventManifest
+    case _: AclEvent          => aclEventManifest
+    case _: RealmEvent        => realmEventManifest
+    case _: OrganizationEvent => organizationEventManifest
+    case _: ProjectEvent      => projectEventManifest
+    case _                    => throw new IllegalArgumentException(s"Unknown event type '${o.getClass.getCanonicalName}'")
+  }
+
+  override def toBinary(o: AnyRef): Array[Byte] = o match {
+    case e: PermissionsEvent  => e.asJson.noSpaces.getBytes(StandardCharsets.UTF_8)
+    case e: AclEvent          => e.asJson.noSpaces.getBytes(StandardCharsets.UTF_8)
+    case e: RealmEvent        => e.asJson.noSpaces.getBytes(StandardCharsets.UTF_8)
+    case e: OrganizationEvent => e.asJson.noSpaces.getBytes(StandardCharsets.UTF_8)
+    case e: ProjectEvent      => e.asJson.noSpaces.getBytes(StandardCharsets.UTF_8)
+    case _                    => throw new IllegalArgumentException(s"Unknown event type '${o.getClass.getCanonicalName}'")
+  }
+
+  override def fromBinary(bytes: Array[Byte], manifest: String): AnyRef = manifest match {
+    case `permissionsEventManifest`  => parseAndDecode[PermissionsEvent](bytes, manifest)
+    case `aclEventManifest`          => parseAndDecode[AclEvent](bytes, manifest)
+    case `realmEventManifest`        => parseAndDecode[RealmEvent](bytes, manifest)
+    case `organizationEventManifest` => parseAndDecode[OrganizationEvent](bytes, manifest)
+    case `projectEventManifest`      => parseAndDecode[ProjectEvent](bytes, manifest)
+    case _                           => throw new IllegalArgumentException(s"Unknown manifest '$manifest'")
+  }
+
+  private def parseAndDecode[E <: Event: Decoder](bytes: Array[Byte], manifest: String): E = {
+    val string = new String(bytes, StandardCharsets.UTF_8)
+    decode[E](string)
+      .getOrElse(throw new IllegalArgumentException(s"Unable to decode for manifest '$manifest' event '$string'"))
+  }
+}
+
+@nowarn("cat=unused")
+object EventSerializer {
+
+  final val permissionsEventManifest: String  = "permissions"
```

Comment:
Why don't you use the names from the modules instead of defining those hardcoded strings here?
`Permissions.moduleType`
`Realms.moduleType`
...